src/Bicep.Core/LanguageConstants.cs

```diff
@@ -38,11 +39,9 @@ public static class LanguageConstants
         public static readonly TypeSymbol Array = new ArrayType("array");
 
         // declares the description property but also allows any other property of any type
-        public static readonly TypeSymbol ParameterModifierMetadata = new NamedObjectType(nameof(ParameterModifierMetadata), CreateParameterModifierMetadataProperties(), Any, TypePropertyFlags.Constant);
+        public static readonly TypeSymbol ParameterModifierMetadata = new NamedObjectType(nameof(ParameterModifierMetadata), CreateParameterModifierMetadataProperties(), new TypeProperty("additionalProperties", Any, TypePropertyFlags.Constant));
```

Comment:
>additionalProperties [](start = 188, length = 20)

The `"additionalProperties"` property name is confusing. Why not just create a new base class for TypeProperty that has all the properties except Name and then have TypeProperty derive from that and add the Name property?

Reply:
The `Constant` part was missing before. That's a good catch!

---
In reply to: [486516659](https://github.com/Azure/bicep/pull/475#discussion_r486516659) [](ancestors = 486516659)